streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -298,6 +299,42 @@ private void closeDirtyAndRevive(final Collection<Task> taskWithChangelogs, fina
         }
     }
 
+    private void commitActiveTasks(final Set<Task> activeTasksNeedCommit, final AtomicReference<RuntimeException> activeTasksCommitException) {
```

Comment:
Could we reuse `commitTasksAndMaybeUpdateCommittableOffsets` instead, to avoid code duplications?

Reply:
Done. I see `commitTasksAndMaybeUpdateCommittableOffsets` already has the `task.commitNeeded()` check amongst other things so it makes all the more sense. Thanks for pointing this out.